### PR TITLE
Fix EC2 test suite to work with testing policies

### PR DESCRIPTION
**Pull request by willthames**
##### SUMMARY
* Update testing policies to ensure all required permissions are present
* Remove iam:CreateRole, as that's not a sensible CI permission
* Make roles static so that they can be present before CI is run,
  meaning that role creation permission is not required by the CI
  itself, only by someone setting up the roles prior to testing
* Remove EFS policy and move contents to storage policy to ensure policy
  count (maximum of 10) stays low

##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
ec2_instance

##### ANSIBLE VERSION

```
ansible 2.7.0.dev0 (devel 3921f34253) last updated 2018/08/20 20:53:37 (GMT +1000)
  config file = None
  configured module search path = [u'/Users/will/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /Users/will/src/ansible/lib/ansible
  executable location = /Users/will/src/ansible/bin/ansible
  python version = 2.7.14 (default, Mar  9 2018, 23:57:12) [GCC 4.2.1 Compatible Apple LLVM 9.0.0 (clang-900.0.39.2)]
```

### CODE REVIEW

**Comment by willthames** [comment]
Rebased on #57716 - can move from WIP after that

File: hacking/aws_config/testing_policies/database-policy.json [hunk 1]
```diff
@@ -78,6 +78,17 @@
           ],
           "Effect": "Allow",
           "Resource": "*"
+        },
+        {
+          "Sid": "DMSEndpoints",
+          "Effect": "Allow",
+          "Action": [
+            "dms:CreateEndpoint",
```
Comment (s-hertel) [inline hacking/aws_config/testing_policies/database-policy.json:94]: Were these intended to be in this PR?

**Review by s-hertel — APPROVED** [review]
Looks good. This is one of the (many) test suites I cannot run locally anymore without making modifications to it. In my pyenv virtualenv `--site-packages` are not picked up since the major test infra restructure. Using system python or docker I get errors like `ERROR! the role 'prepare_tests' was not found in ...`. The latter goes away if I add a meta/main.yml with the dependencies, but I'm not sure what is unique about my environment that other people don't seem to have the same problems. Anyway, tests pass with modification to get them to run.

File: hacking/aws_config/testing_policies/database-policy.json (hunk 1 above)
Comment (willthames) [reply hacking/aws_config/testing_policies/database-policy.json:94]: I'm not really sure how they came to be in this PR but given that there are DMS tests, and these are the only dms privileges in the testing policies, I'm favouring leaving them in

**Comment by willthames** [comment]
@s-hertel I don't think that's weird - I can't get the runme.sh tests to work - I thought I'd run these since the rebase but clearly not.

I'll tidy this up more to move from runme.sh to the same approach as ecs_cluster.

Apologies, I thought this was readier than it is.

**Comment by s-hertel** [comment]
@willthames Ah, I'm glad to not be alone, been feeling crazy. :-)

File: hacking/aws_config/testing_policies/database-policy.json (hunk 1 above)
Comment (jillr) [reply hacking/aws_config/testing_policies/database-policy.json:94]: This would eliminate the need for the policy creation in #56980 then, right?

Comment (willthames) [reply hacking/aws_config/testing_policies/database-policy.json:94]: @jillr yes, they wouldn't be needed there

File: hacking/aws_config/testing_policies/security-policy.json [hunk 2]
```diff
@@ -91,22 +91,6 @@
                 "arn:aws:logs:{{aws_region}}:{{aws_account}}:log-group:ansible-testing*"
             ]
         },
-        {
-            "Sid": "AllowSTSAnsibleTests",
-            "Action": [
-                "iam:CreateRole",
-                "iam:DeleteRole",
-                "iam:DetachRolePolicy",
-                "sts:AssumeRole",
```
Comment (s-hertel) [inline hacking/aws_config/testing_policies/security-policy.json:100]: The other permissions being removed here seem to be present in the rest of the file, except this one. It seems to be at least used by the sts_assume_role tests.

File: hacking/aws_config/testing_policies/security-policy.json [hunk 3]
```diff
@@ -132,26 +116,16 @@
...
         {
-            "Sid": "AllowAccessToSpecifiedIAMResources",
+            "Sid": "AllowAccessToServerCertificates",
             "Effect": "Allow",
             "Action": [
-                "iam:CreateRole",
-                "iam:DeleteRole",
-                "iam:GetRole",
-                "iam:PassRole",
-                "iam:UpdateAssumeRolePolicy"
```
Comment (s-hertel) [inline hacking/aws_config/testing_policies/security-policy.json:142]: Maybe not likely, but it seems possible if a test uses a static role that isn't cleaned up at the end of a test run and it is modified, then rerunning the tests would try to use this permission. It doesn't look like it's duplicated anywhere in the policies.

File: hacking/aws_config/testing_policies/security-policy.json [hunk 4]
```diff
@@ -132,26 +116,16 @@
...
+                "iam:ListServerCertificates",
+                "iam:UploadServerCertificate",
```
Comment (s-hertel) [inline hacking/aws_config/testing_policies/security-policy.json]: I don't think we have any modules that make the following several API calls.

**Review by s-hertel — COMMENTED** [review]
Looks good and tests pass.

**Pull request merged**